components/binary_sensor/xiaomi.py

```diff
@@ -57,25 +65,29 @@ def is_on(self):
         """Return true if sensor is on."""
         return self._state
 
+    @property
+    def device_state_attributes(self):
+        attrs = {ATTR_NO_MOTION_SINCE: self._no_motion_since}
+        attrs.update(super().device_state_attributes)
+        return attrs
+
     def parse_data(self, data):
-        if 'no_motion' in data:
-            if self._state:
-                self._state = False
-                return True
-            else:
-                return False
+        if NO_MOTION in data:  # handle push from the hub
+            self._no_motion_since = data[NO_MOTION]

```

Comment:
I think we should keep:

```
-            if self._state:
 -                self._state = False
 -                return True
 -            else:
 -                return False
```
This will make sure we only push an update to HA if there is a change in the state